include/alpaka/acc/Tag.hpp

```diff
@@ -0,0 +1,64 @@
+/* Copyright 2022 Simeon Ehrig
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <iostream>
+#include <type_traits>
+
+#define CREATE_ACC_TAG(tag_name)                                                                                      \
+    struct tag_name                                                                                                   \
+    {                                                                                                                 \
+        static std::string get_name()                                                                                 \
+        {                                                                                                             \
+            return #tag_name;                                                                                         \
+        }                                                                                                             \
+    }
+
+namespace alpaka
+{
+    CREATE_ACC_TAG(AccCpuFibersTag);
+    CREATE_ACC_TAG(AccCpuOmp2BlocksTag);
+    CREATE_ACC_TAG(AccCpuOmp2ThreadsTag);
+    CREATE_ACC_TAG(AccCpuSerialTag);
+    CREATE_ACC_TAG(AccCpuSyclIntelTag);
+    CREATE_ACC_TAG(AccCpuTbbBlocksTag);
+    CREATE_ACC_TAG(AccCpuThreadsTag);
+    CREATE_ACC_TAG(AccFpgaSyclIntelTag);
+    CREATE_ACC_TAG(AccFpgaSyclXilinxTag);
+    CREATE_ACC_TAG(AccGenericSyclTag);
+    CREATE_ACC_TAG(AccGpuCudaRtTag);
+    CREATE_ACC_TAG(AccGpuHipRtTag);
+    CREATE_ACC_TAG(AccGpuSyclIntelTag);
+    CREATE_ACC_TAG(AccOaccTag);
+    CREATE_ACC_TAG(AccOmp5Tag);
+
+    template<typename TAcc>
+    struct AccToTag;
+
+    template<typename TAcc>
+    using AccToTag_t = typename AccToTag<TAcc>::type;
+
+    template<typename TTag, typename TDim, typename TIdx>
+    struct TagToAcc;
+
+    template<typename TTag, typename TDim, typename TIdx>
+    using TagToAcc_t = typename TagToAcc<TTag, TDim, TIdx>::type;
+
+    // TODO: make variadic, to support more than on tag per call
+    template<typename TAcc, typename TTag>
+    using is_acc_tag_match = std::is_same<AccToTag_t<TAcc>, TTag>;
+
+    template<typename TAcc, typename TTag>
+    inline constexpr bool is_acc_tag_match_v = alpaka::is_acc_tag_match<TAcc, TTag>::value;
```

Comment:
Variadic version:
```suggestion
    template<typename TAcc, typename... TTags>
    inline constexpr bool is_acc_tag_match_v = (alpaka::is_acc_tag_match<TAcc, TTag>::value || ...);
```
And again, I would rename it to not have the `_v` and use `CamelCase`:
```suggestion
    template<typename TAcc, typename... TTags>
    inline constexpr bool AccMatchesTags = (alpaka::is_acc_tag_match<TAcc, TTag>::value || ...);
```

Reply:
Thanks for the variadic version. Saves me time. @psychocoderHPC means, that is maybe required to have a type and value for some traits.

Reply:
Wait, it should be lowercase, since it is a variable name, so `accMatchesTags`.

Reply:
Renamed it after I found the reaction of @psychocoderHPC . 
@psychocoderHPC  You should not answer with a reaction. I didn't get a notification. It was only luck, that I saw it.